File: Typescript/bubble_sort.ts
function bubbleSort(array: number[]): number[] {
  console.log("Before Sotr: ", array);
  for (let i = 0; i < array.length; i++) {
    for (let j = 0; j < array.length - 1; j++) {
      if (array[j] > array[j + 1]) {
        let tmp = array[j];
        array[j] = array[j + 1];
        array[j + 1] = tmp;
      }
    }
  }
  console.log("After Sort: ", array);
  return array;
}

bubbleSort([20, 15, 36, 45, 2, 9, 0]);

File: Dart/BubbleSort.dart
void main(){
  var list = [6,7,4,2,3]; // declaring list
  int n,temp;
  
  print("Array Before Sorting :");
   for (int i=0 ;i < 5;i++) {
     print(list[i]); // printing list elements
	}
  n = list.length;  // .length will return length of list

  // Bubble Sort
  for (int i=0 ;i < n-1;i++) {
    for(int j=0;j< n-i-1;j++){
      if(list[j] > list[j+1]){
        temp = list[j];
        list[j] = list[j+1];
        list[j+1] = temp;
      }
    }  	
  }
   print("Array After BubbleSort :");
    for (int i=0 ;i < 5;i++) {
     print(list[i]);
	}
}

